pandas/core/indexes/datetimes.py

```diff
@@ -1730,9 +1731,13 @@ def offset(self):
     def offset(self, value):
         """get/set the frequency of the Index"""
         msg = ('DatetimeIndex.offset has been deprecated and will be removed '
-               'in a future version; use DatetimeIndex.freq instead.')
+               'in a future version; use DatetimeIndex.set_freq instead.')
         warnings.warn(msg, FutureWarning, stacklevel=2)
-        self.freq = value
+        if value is not None:
+            value = to_offset(value)
```

Comment:
call set_freq here

Reply:
This is for the deprecated `offset` setter which is just an alias for the `freq` setter, so I don't think `set_freq` should be used here since setting is to happen inplace (as was the previous convention).  This previously would call the `freq` setter, but since that's also being deprecated it was subsequently raising the `freq` setter warning too, so I just reused the `freq` setter code to avoid the additional warning.